src/ipc-handlers.ts

```diff
@@ -571,20 +570,16 @@ export async function saveOnboarding(
 
 export async function executeWPCLiInline(
 	_event: IpcMainInvokeEvent,
-	{ projectPath, args }: { projectPath: string; args: string }
-) {
-	const wpCliArgs = parse( args );
-
-	// The parsing of arguments can include shell operators like `>` or `||` that the app don't support.
-	const isValidCommand = wpCliArgs.every(
-		( arg: unknown ) => typeof arg === 'string' || arg instanceof String
-	);
-	if ( ! isValidCommand ) {
-		throw Error( `Can't execute wp-cli command with arguments: ${ args }` );
+	{ siteId, args }: { siteId: string; args: string }
+): Promise< WpCliResult > {
+	if ( SiteServer.isDeleted( siteId ) ) {
+		return { stdout: '', stderr: `can't execute command on deleted site ${ siteId }` };
```

Comment:
Where does this `stderr` value surface? Is it presented to the user? If so, outputting the site ID may not be all that helpful as we do not surface that information/concept to users anywhere else. 

Suggestion to improve capitalization and increase clarity through avoiding contractions. Feel free to disregard. 
```suggestion
		return { stdout: '', stderr: `Cannot execute command on deleted site ${ siteId }` };
```

Reply:
> Where does this `stderr` value surface? Is it presented to the user? If so, outputting the site ID may not be all that helpful as we do not surface that information/concept to users anywhere else.

This error should only be generated by `wp-cli` commands invoked from the AI chat context ([reference](https://github.com/Automattic/studio/blob/666f7ea859b354136761dcc32ac3d8d3ff690d6e/src/hooks/use-chat-context.tsx#L75-L95)). The error descriptor is not surfaced to the user in that case, so this message will be mainly an internal note. However, the error descriptor is shown to the user in commands executed within the AI Assistant chat. But as I mentioned before, it's unlikely that those invocations result in this error.

I agree that site ID won't be helpful to debug a potential issue, so we could use site name instead. WDYT?

Reply:
> I agree that site ID won't be helpful to debug a potential issue, so we could use site name instead. WDYT?

I've just realized that we only keep the site ID for deleted sites. We could expand the deleted sites list with more information like the site's name, however, since this information won't really be exposed to the user maybe it's not worth it. WDYT?

Reply:
I suppose it comes down to how frequent and disruptive this error may be. My perception is that it is infrequent and low disruption. If that is true, then there is likely little value in making this more robust. 

Do you perceive the error similarly?

Reply:
Yep, this error won't be seen by the user so we can keep it as-is 👍.